### DISCUSSION TITLE
Installation without internet access (organisation security)

### DISCUSSION THREAD

**Discussion by SebHonorine** [DependencyTrack/dependency-track#1259] [Q&A]
Hello there,

I saw https://github.com/DependencyTrack/dependency-track/issues/717 is this still the case in 2021 ? 

We are creating a cluster AKS with Owasp-DT and CAN'T have internet access for security to get CVE database. How, can I get solution for my situation. Is it Possible to install then configure DB with an outsourced database ?

thank you.

**Comment by stevespringett** [comment]
DT mirrors the NVD, but that can be worked around so that DT can use it locally. The problem is the NVD is still relying solely on CPE, so it will be impossible to identify vulnerabilities in the majority of software that is consumed during build-time operations. You'll only be able to identify vulnerabilities in components that have a CPE defined.

VulnDB, a commercial service from RBS, has support for purl, and there is the ability to download their entire catalog - but it does take thousands of REST requests to do so.

OSS Index doesn't provide offline capabilities. DT will be integrating with Snyk soon, but their APIs require Internet access. So for airgapped environments, the quantity and quality of vulnerability intelligence sources is severely limited.

DT does have an Internal analyzer. Its the same analyzer used for CPE matching for the NVD. The Internal analyzer supports purl, but I don't know if its been fully built out yet. It may work, it may not. When the private vulnerability repo support makes its way into DT, it will be fully supported.

**Reply by SebHonorine** [reply]
Thank you for the answer Steve. 
So yeah it's possible to get some of the vulnerabilities. But, ultimately you want to be able to have internet access.

**Reply by SebHonorine** [reply]
Sorry another question linked.
Can we install Owasp DT on a pod close to internet while having the Database which contain the Vulnerabilities hosted on a pod open to internet to get the update etc ?

**Reply by stevespringett** [reply]
You dont want two instances of DT writing to the same database. You'd also have similar issues with only being able to check for vulnerabilities using CPE.
